sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/ApplyFunctionExpression.scala

```diff
@@ -29,11 +29,12 @@ case class ApplyFunctionExpression(
   override def name: String = function.name()
   override def dataType: DataType = function.resultType()
 
+  private lazy val childrenWithIndex = children.zipWithIndex
   private lazy val reusedRow = new GenericInternalRow(children.size)
 
   /** Returns the result of evaluating this expression on a given input Row */
   override def eval(input: InternalRow): Any = {
-    children.zipWithIndex.foreach {
+    childrenWithIndex.foreach {
```

Comment:
This file change is orthogonal to this PR. Can we split this change as a separate PR?

Reply:
+1

Reply:
Sure will do.